pkg/compare/compare.go

```diff
@@ -146,6 +151,59 @@ func UnmarshalInlineYaml(obj map[string]interface{}, targetPath string) (err err
 	return
 }
 
+// genPathIgnoreOpt returns a cmp.Option to ignore paths specified in parameter ignorePaths.
+func genYamlIgnoreOpt(yamlStr string) (cmp.Option, error) {
+	tree := make(map[string]interface{})
+	if err := yaml.Unmarshal([]byte(yamlStr), &tree); err != nil {
+		return nil, err
+	}
+	return cmp.FilterPath(func(curPath cmp.Path) bool {
+		up := pathToStringList(curPath)
+		found, _ := IsPathInTree(tree, up)
+		return found
+	}, cmp.Ignore()), nil
+}
+
+func IsPathInTree(tree map[string]interface{}, path []string) (bool, error) {
```

Comment:
does it need to be public?

Reply:
Removed this func since this algorithm does not mutate path.

Reply:
And change it into private.